src/state.rs

```diff
@@ -365,78 +370,126 @@ impl TabExpandedString {
     }
 }
 
-/// Estimate the number of seconds per step
+/// Estimate the number of steps per second
+///
+/// Exponentially smoothed estimate with an adjustable weighting
```

Comment:
Would be good to comment a bit more on the how and why of the double smoothing trick.